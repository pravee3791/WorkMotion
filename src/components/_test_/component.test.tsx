import React from 'react';
import ReactDom from "react-dom";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Status from "../status/status";


afterEach(cleanup);

it('reders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render<any, any>(<Status></Status>, div)
})

it('checks if status exists', () => {
    const StatusPage = render(<Status />);
    const statusList = StatusPage.container.querySelector('#status-wrapper-container');
    expect(statusList).toBeTruthy();
    expect(statusList).toBeDefined();
})

it(' component matches snapshot', () => {
    const StatusPage = render(<Status />);
    const tree = renderer.create(<Status></Status>).toJSON();
    expect(tree).toMatchSnapshot();
})
