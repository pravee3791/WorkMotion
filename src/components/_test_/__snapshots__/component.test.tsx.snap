// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` component matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="arrow-steps clearfix"
      id="status-wrapper-container"
    >
      <div
        className="step"
        id={0}
        onClick={[Function]}
      >
         
        <span>
           
          ADDED
        </span>
         
      </div>
      <div
        className="step"
        id={1}
        onClick={[Function]}
      >
         
        <span>
           
          IN-CHECK
        </span>
         
      </div>
      <div
        className="step"
        id={2}
        onClick={[Function]}
      >
         
        <span>
           
          APPROVED
        </span>
         
      </div>
      <div
        className="step"
        id={3}
        onClick={[Function]}
      >
         
        <span>
           
          ACTIVE
        </span>
         
      </div>
      <div
        className="step"
        id={4}
        onClick={[Function]}
      >
         
        <span>
           
          INACTIVE
        </span>
         
      </div>
    </div>
  </div>
</div>
`;
